chia/util/augmented_chain.py

```diff
@@ -44,11 +45,17 @@ def add_extra_block(self, block: FullBlock, block_record: BlockRecord) -> None:
         assert block.header_hash == block_record.header_hash
         self._extra_blocks[block_record.header_hash] = (block, block_record)
         self._height_to_hash[block_record.height] = block_record.header_hash
+        if self._peak_height is None:
+            self._peak_height = block.height
+        else:
+            self._peak_height = max(block.height, self._peak_height)
 
     def remove_extra_block(self, hh: bytes32) -> None:
         if hh in self._extra_blocks:
             block_record = self._extra_blocks.pop(hh)[1]
```

Comment:
@arvidn lmk know what you think about this, i think that enforcing that we only ever remove the top and not remove blocks in the middle of the chain makes sense